plutus-ledger/src/Ledger/Scripts.hs

```diff
@@ -84,10 +86,22 @@ import           LedgerBytes                         (LedgerBytes (..))
 -- | A script on the chain. This is an opaque type as far as the chain is concerned.
 newtype Script = Script { unScript :: UPLC.Program UPLC.DeBruijn PLC.DefaultUni PLC.DefaultFun () }
   deriving stock Generic
-  deriving Serialise via UPLC.OmitUnitAnnotations UPLC.DeBruijn PLC.DefaultUni PLC.DefaultFun
+  deriving newtype Flat
 -- | Don't include unit annotations in the CBOR when serialising.
 -- See Note [Serialising Scripts] in Language.PlutusCore.CBOR
 
+{-| A lot of the plutus infrastructure uses CBOR for serialisation, but we are using
+  flat for encoding plutus terms. Here we are providing a CBOR instance for plutus
+  terms by encoding the term to a `ByteString` value using flat, and further encode
+  it using CBOR. -}
+instance Serialise Script where
```

Comment:
This instance will use the `Flat` encoding to generate a `ByteString` which then is encoded by `CBOR`. This is where we switch from PLC Flat to PAB CBOR.